systemtest/src/test/java/io/strimzi/systemtest/operators/user/UserST.java

```diff
@@ -216,6 +223,85 @@ void testUserWithQuotas(KafkaUser user) {
         });
     }
 
+    @Test
+    void testCreatingUsersWithSecretPrefix() {
+        String clusterName = "second-cluster";
+        String secretPrefix = "top-secret-";
+        String tlsUserName = "encrypted-leopold";
+        String scramShaUserName = "scramed-leopold";
+
+        KafkaResource.kafkaEphemeral(clusterName, 3)
+            .editSpec()
+                .editKafka()
+                    .withNewListeners()
+                        .addNewGenericKafkaListener()
+                            .withName(Constants.TLS_LISTENER_DEFAULT_NAME)
+                            .withPort(9093)
+                            .withType(KafkaListenerType.INTERNAL)
+                            .withTls(true)
+                            .withNewKafkaListenerAuthenticationTlsAuth()
+                            .endKafkaListenerAuthenticationTlsAuth()
+                        .endGenericKafkaListener()
+                    .endListeners()
+                .endKafka()
+                .editEntityOperator()
+                    .editUserOperator()
+                        .withNewSecretPrefix(secretPrefix)
+                    .endUserOperator()
+                .endEntityOperator()
+            .endSpec()
+            .done();
+
+        KafkaTopicResource.topic(clusterName, TOPIC_NAME).done();
+        KafkaUser tlsUser = KafkaUserResource.tlsUser(clusterName, tlsUserName).done();
+        KafkaUserResource.scramShaUser(clusterName, scramShaUserName).done();
+
+        KafkaClientsResource.deployKafkaClients(true, clusterName + "-" + Constants.KAFKA_CLIENTS, true, null, secretPrefix, tlsUser).done();
+        String kafkaClientsName = kubeClient().listPodsByPrefixInName(clusterName + "-" + Constants.KAFKA_CLIENTS).get(0).getMetadata().getName();
+
+        InternalKafkaClient internalKafkaClient = new InternalKafkaClient.Builder()
+            .withUsingPodName(kafkaClientsName)
+            .withNamespaceName(NAMESPACE)
+            .withTopicName(TOPIC_NAME)
+            .withKafkaUsername(tlsUserName)
+            .withSecurityProtocol(SecurityProtocol.SASL_SSL)
+            .withListenerName(Constants.TLS_LISTENER_DEFAULT_NAME)
+            .withClusterName(clusterName)
+            .withMessageCount(MESSAGE_COUNT)
+            .withSecretPrefix(secretPrefix)
+            .build();
+
+        internalKafkaClient.assertSentAndReceivedMessages(
+            internalKafkaClient.sendMessagesTls(),
+            internalKafkaClient.receiveMessagesTls()
+        );
```

Comment:
Should you try to send/receive also for the SCRAM user?

Reply:
Yes, I wanted to add it :) thanks